Source/Dafny/AST/DafnyAst.cs

```diff
@@ -6927,8 +6927,35 @@ public Statement(IToken tok, IToken endTok)
 
     /// <summary>
     /// Returns the non-null expressions of this statement proper (that is, do not include the expressions of substatements).
+    /// Filters all sub expressions that are not part of specifications
     /// </summary>
-    public virtual IEnumerable<Expression> SubExpressions {
+    public IEnumerable<Expression> SubExpressions {
```

Comment:
I would expect us to have all error messages of the form "...is not compilable..." to occur after a de-ghosting phase in the compiler pipeline, after which all Expressions should be non-specification/ghost expressions, so I'm a bit surprised we need the change this PR makes, but I guess without the current architecture we need it.